src/classic/clvm_tools/stages/stage_2/optimize.rs

```diff
@@ -620,19 +664,34 @@ impl<'a> OptimizerRunner<'a> {
 
 pub fn optimize_sexp_<'a>(
     allocator: &mut Allocator,
+    memo: Rc<RefCell<HashMap<String, NodePtr>>>,
     r_: NodePtr,
     eval_f: Rc<dyn TRunProgram>,
 ) -> Result<NodePtr, EvalErr> {
     let mut r = r_;
+    let footprint = sha256tree(allocator, r).hex();
+
+    {
+        let memo_rc: &RefCell<HashMap<String, NodePtr>> = memo.borrow();
+        let memo_ref: Ref<HashMap<String, NodePtr>> = memo_rc.borrow();
+        let memo: &HashMap<String, NodePtr> = memo_ref.borrow();
```

Comment:
I think it would look better to just say `memo.borrow().borrow().borrow()`. But I'm also suspicious that there might be ways to make this a simpler data type

Reply:
rust wouldn't be able to narrow down what type the inner ones were.
even a single level it may not be possible without an annotation.
my habit is generally to call out the names, but i suppose it isn't a popular choice.